### PR TITLE
[pkp-lib] #6925 Change Native Import/Export plugin so that it takes the user from the Application and not the command

### PR DESCRIPTION
**Pull request by defstat**

### CODE REVIEW

File: classes/cliTool/CliTool.inc.php [hunk 1]
```diff
@@ -81,4 +89,56 @@ function __construct($argv = array()) {
 	function usage() {
 	}
 
+	private function checkArgsForUsername() {
+		$usernameKeyPos = array_search('--user_name', $this->argv);
```
Comment (NateWr) [inline classes/cliTool/CliTool.inc.php:93]: Can we support `--user_name` and `-u`? Just to make it a little quicker...

File: classes/cliTool/CliTool.inc.php [hunk 2]
```diff
@@ -81,4 +89,56 @@ function __construct($argv = array()) {
...
+		}
+
+		if (!$this->user) {
+			$userGroupDao = DAORegistry::getDAO('UserGroupDAO'); /** @var $userGroupDao UserGroupDAO */
+			$adminGroups = $userGroupDao->getUserGroupIdsByRoleId(ROLE_ID_SITE_ADMIN);
+
+			if (count($adminGroups)) {
+				$groupUsers = $userGroupDao->getUsersById($adminGroups[0])->toArray();
+
+				$this->setUser($groupUsers[0]);
```
Comment (NateWr) [inline classes/cliTool/CliTool.inc.php:123]: Maybe double-check if the user exists and `exit(0)` if not?

File: classes/cliTool/CliTool.inc.php [hunk 3]
```diff
@@ -81,4 +89,56 @@ function __construct($argv = array()) {
...
+		$registeredUser = Registry::get('user', true, null);
+		if (!isset($registeredUser)) {
+			/**
+			 * This is used in order to reconcile with possible $request->getUser()
+			 * used inside import processes, when the import is done by CLI tool.
+			 */
+			if ($user) {
+
+			}
+			Registry::set('user', $user);
```
Comment (NateWr) [inline classes/cliTool/CliTool.inc.php:138]: I think this needs to be in the `if` block above, right?

File: classes/cliTool/CliTool.inc.php [hunk 4]
```diff
@@ -81,4 +89,56 @@ function __construct($argv = array()) {
...
+		}
+
+		$this->user = $user;
```
Comment (NateWr) [inline classes/cliTool/CliTool.inc.php:141]: Should this be `$registeredUser` if the conditional blocks above are `false`?

**Review by NateWr — CHANGES_REQUESTED** [review]
Looks good! :+1: Just a couple minor comments.

File: classes/cliTool/CliTool.inc.php (hunk 1 above)
Comment (defstat) [reply classes/cliTool/CliTool.inc.php:92]: Fixed

File: classes/cliTool/CliTool.inc.php (hunk 2 above)
Comment (defstat) [reply classes/cliTool/CliTool.inc.php]: Fixed

File: classes/cliTool/CliTool.inc.php (hunk 3 above)
Comment (defstat) [reply classes/cliTool/CliTool.inc.php]: Fixed

File: classes/cliTool/CliTool.inc.php (hunk 4 above)
Comment (defstat) [reply classes/cliTool/CliTool.inc.php]: Fixed

File: classes/cliTool/CliTool.inc.php [hunk 5]
```diff
@@ -72,13 +78,84 @@ function __construct($argv = array()) {
...
+
+			if (count($adminGroups)) {
+				$groupUsers = $userGroupDao->getUsersById($adminGroups[0])->toArray();
+
+				if (count($groupUsers) > 0) {
+					$this->setUser($groupUsers[0]);
+				} else {
+					$this->exitWithUsageMessage();
+				}
+			}
```
Comment (NateWr) [inline classes/cliTool/CliTool.inc.php:129]: If this condition fails: `if (count($adminGroups))`, then no user could be set and the CLI should exit with the usage message, right?

Comment (defstat) [reply classes/cliTool/CliTool.inc.php:133]: @NateWr are you confirming that this is intentional? Even though not all CLI tools need a user to run their code, the initial case that this PR is trying to fix, is the one where the CLI tool code makes, for example, an service call, that in turn needs a user defined in order to be executed. So, this PR tries to make the user mandatory for CLI tools.

Comment (NateWr) [reply classes/cliTool/CliTool.inc.php:133]: That's a good point. I guess we shouldn't enforce this yet. So leave it as-is. :+1:

**Pull request merged**